CHANGELOG.d/feature_optimize-uncurried-st.md

```diff
@@ -0,0 +1 @@
+* Adds a compiler optimization for `ST` functions with up to 10 arity, similar to `Effect` optimizations.
```

Comment:
Change ‘Adds’ to ‘Add’, please? The changelog uses the same imperative mood as is recommended for Git commit messages.